src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1536,6 +1555,7 @@ private function addDefaultParametersMethod(): string
 
     public function getParameter(string $name): array|bool|string|int|float|\UnitEnum|null
     {
+        __trigger_deprecation__
```

Comment:
This is not consistent with the ParameterBag logic, where a parameter is first checked for existence and only then for deprecation

Reply:
Fixed, thanks!